crates/subspace-fraud-proof/src/lib.rs

```diff
@@ -252,11 +241,16 @@ where
             ..
         } = proof;
 
+        let at = PBlock::Hash::decode(&mut parent_hash.encode().as_slice())
```

Comment:
If moving here resolves an issue, it seems to be like a race condition where transaction that generates runtime upgrade may not go through yet. I'd really like to figure out what is going on instead as we might hit the same issue under different circumstances and it feels important.

Reply:
I don't think runtime upgrade is related because there is no runtime upgrade at all in the test. And we are using calling a runtime API which is called on top of the state that the certain block has been applied, the runtime upgrade must have been done at that point if there is one.